protocol/messenger_chats.go

```diff
@@ -13,15 +13,38 @@ func (m *Messenger) Chats() []*Chat {
 	var chats []*Chat
 
 	m.allChats.Range(func(chatID string, chat *Chat) (shouldContinue bool) {
-		chats = append(chats, chat)
+		if chat.Active {
+			if chat.LastMessage != nil {
+				text := ""
+				if chat.LastMessage.ContentType == protobuf.ChatMessage_TEXT_PLAIN {
+					if len(chat.LastMessage.Text) > 200 {
+						text = chat.LastMessage.Text[:200]
+					}
+					text = chat.LastMessage.Text
+				}
+				chat.LastMessage = &common.Message{
+					ChatMessage: protobuf.ChatMessage{
+						ContentType: chat.LastMessage.ContentType,
+						Text:        text},
+				}
+			}
+
+			chat.Members = nil
```

Comment:
This is a bit dangerous, 
I think you are resetting the field for good, as we use a pointer, so it would be gone from memory, I think what you want to do is to use a custom marshaler for chats, where you can do this, or make a copy of the chat (probably neater to use a custom marshaler)

Reply:
thanks, created new ChatPreview struct